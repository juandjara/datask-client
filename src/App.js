import React, { Component } from 'react';
import RaisedButton from 'material-ui/RaisedButton';
import logo from './logo.svg';
import './App.css';

const buttonStyle = {
  margin: ".5em"
}

class App extends Component {
  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcome to React</h2>
        </div>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        <RaisedButton style={buttonStyle} label="Default" />
        <RaisedButton style={buttonStyle} primary={true} label="Primary" />        
        <RaisedButton style={buttonStyle} secondary={true} label="Secondary" />
      </div>
    );
  }
}

export default App;
