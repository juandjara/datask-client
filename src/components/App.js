import React, { Component } from 'react'
import Layout from 'react-toolbox/lib/layout/Layout';
import Panel from 'react-toolbox/lib/layout/Panel';
import Icon from 'react-toolbox/lib/font_icon/FontIcon'
import ReactTooltip from 'react-tooltip';
import Sidenav from './Sidenav';
import Header from './Header';
import TaskQuickAccess from './TaskQuickAccess'
import TimeCounters from './TimeCounters'
import { connect } from 'react-redux'
import { toggleSidenavOpen } from '../ducks/sidenav'

class App extends Component {
  handleToggle = () => this.props.dispatch(toggleSidenavOpen());
  render() {
    const { sidenav, children } = this.props;
    const containerStyle = {};
    if (sidenav.pinned && sidenav.open) {
      containerStyle.paddingLeft = 240
    }
    return (
      <Layout>
        <Sidenav />
        <Panel style={containerStyle}>
          <ReactTooltip place="right" effect="solid" />         
          <Header onToggleSidenav={this.handleToggle} />
          <div style={{marginTop: 56}}>
            <div style={{display: 'flex'}}>
              <section style={{flex: 1, marginRight: '2px', marginBottom: '.5rem'}}>
                <p style={{display: 'flex', marginBottom: '.75rem', marginTop: '1.5rem'}} >
                  <Icon className="task-cards-star">star</Icon>
                  Tareas destacadas
                </p>
                <TaskQuickAccess />
              </section>
              <div style={{position: 'fixed', right: 0}}>
                <TimeCounters />
              </div>
            </div>
            <main className="main">{children}</main>
          </div>
        </Panel>
      </Layout>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    sidenav: state.sidenav
  }
}

export default connect(mapStateToProps)(App)