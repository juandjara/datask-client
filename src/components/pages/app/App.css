.main > * {
  max-width: 960px;
  margin: 0 auto;
}

.divider {
  margin: .5em 0;
  min-height: 26px;
  border-top: 1px solid #e4e4e4;
  border-bottom: 1px solid transparent;
  border-left: 1px solid #e4e4e4;
}

.hidden {
  display: none !important;
}

.subheader {
  font-size: 14px;
  color: #444;
}

.truncate-line {
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  display: inline-block;
}

.below-navbar {
  margin-top: 4rem;
}
@media(max-width: 600px) {
  .below-navbar {
    margin-top: 3rem;
  }
}

.toastify-content {
  background: #333 !important;
}

.list-container {
  padding: .5em;
}
.list-title-container {
  margin: 1rem;
}
.list-title {
  margin: 1rem 0;
}
.list-corner-fab {
  position: absolute;
  top: .75em;
  right: 1em;
}
.list {
}
.list-item {
  border-bottom: 1px solid #ccc;
  padding: 5px 0;
}
.list-item:last-child {
  border-bottom: none;
}


.edit-dialog {
  overflow: auto;
  width: 100%;
  max-width: 768px;
  margin: 0 2em;
}

.edit-dialog-active {
  margin: 1em;
}
.edit-dialog-active [data-react-toolbox="label"] {
  padding-left: 2.5em;
}
.edit-dialog-button {
  margin: .5em;
}

.select .Select-control {
  border-top: none;
  border-left: none;
  border-right: none;
  border-radius: 0;
}
.select {
  padding-bottom: 20px;
}
.select-label {
  font-size: 12px;
  color: rgba(0,0,0, .3);
}
.select-icon {
  display: block;
  padding-top: 20px;
  padding-left: 12px;
  padding-right: 32px;
}
.select .Select-value {
  padding-left: 0 !important;
}
.select .Select-input {
  padding-left: 0 !important;
}

.select.is-focused .Select-control {
  border-color: var(--palette-teal-500);
}
.select .Select-menu-outer {
  top: 40px !important;
}
.select-outer-top .Select-menu-outer {
  top: unset !important;
  bottom: 60px !important;
  border-radius: 0px !important;
}

.date-picker-container {
  display: flex;
}
.date-picker-container [data-react-toolbox="date-picker"] {
  flex: 1;
}
