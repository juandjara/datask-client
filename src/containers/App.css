.main > * {
  max-width: 960px;
}

.divider {
  margin: .5em 0;
  min-height: 26px;
  border-top: 1px solid #e4e4e4;
  border-bottom: 1px solid transparent;
  border-left: 1px solid #e4e4e4;
}

.hidden {
  display: none !important;
}

.subheader {
  font-size: 14px;
  color: #444;
}

.truncate-line {
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  display: inline-block;
}

.below-navbar {
  margin-top: 4rem;
}
@media(max-width: 600px) {
  .below-navbar {
    margin-top: 3rem;
  }
}

.toastify-content {
  background: #333 !important;
}

.list-container {
  padding: .5em;
}
.list-title-container {
  margin: 1rem;
}
.list-title {
  margin: 1rem 0;
}
.list-corner-fab {
  position: absolute;
  top: .75em;
  right: 1em;
}
.list {
  padding: 0;
  background-color: white;
  box-shadow: var(--shadow-2p);
}
