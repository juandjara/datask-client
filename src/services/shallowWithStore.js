// https://medium.com/@visualskyrim/test-your-redux-container-with-enzyme-a0e10c0574ec
import { shallow } from 'enzyme';

const shallowWithStore = (component, store) => {
  const context = {
    store,
  };
  return shallow(component, { context });
};

export default shallowWithStore;