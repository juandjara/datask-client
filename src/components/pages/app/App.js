import React, { Component } from 'react'
import Layout from 'react-toolbox/lib/layout/Layout';
import Panel from 'react-toolbox/lib/layout/Panel';
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.min.css';

import {Header} from 'components/shared/header'
import {Sidenav} from 'components/shared/sidenav';
import ShowOnMedia from 'components/shared/ShowOnMedia'
import {TaskQuickAccess} from 'components/shared/taskQuickAccess'

import {fetchProfile} from 'reducers/profile.reducer'
import {
  actions as timeActions,
  selectors as timeSelectors
} from 'reducers/time.reducer'
import {toggleSidenavOpen} from 'reducers/sidenav.reducer'
import {logout} from 'reducers/auth.reducer'

import './App.css';

export class App extends Component {
  componentDidMount() {
    this.props.actions.fetchProfile()
    this.timer = setInterval(
      () => this.props.timeActions.tick(),
      1000
    )
    this.props.timeActions.fetchByUser(this.props.userId, {page: 0})
  }
  componentWillUnmount() {
    clearInterval(this.timer)
  }

  render() {
    const { profile, sidenav, actions, children } = this.props;
    const containerStyle = {
      minHeight: '100vh'
    };
    if (sidenav.pinned) {
      containerStyle.paddingLeft = 256
    }
    return (
      <Layout>
        <Sidenav
          actions={actions}
          profile={profile}
          sidenav={sidenav}
        />
        <Panel style={containerStyle}>
          <ToastContainer autoClose={3000} position="bottom-right" />
          <Header toggleSidenav={actions.toggleSidenavOpen} />
          <ShowOnMedia mediaKey="small">
            <TaskQuickAccess />
          </ShowOnMedia>
          <main className="main">{children}</main>
        </Panel>
      </Layout>
    )
  }
}

const mapStateToProps = (state) => ({
  userId: state.auth._id,
  sidenav: state.sidenav,
  profile: state.profile,
  times: timeSelectors.getByUserId(state, state.auth._id)
})
const mapDispatchToProps = dispatch => ({
  timeActions: bindActionCreators(timeActions, dispatch),
  actions: bindActionCreators({
    logout, 
    fetchProfile, 
    toggleSidenavOpen, 
  }, dispatch)
})

export default connect(mapStateToProps, mapDispatchToProps)(App)
