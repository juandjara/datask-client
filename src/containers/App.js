import React, { Component } from 'react'
import Layout from 'react-toolbox/lib/layout/Layout';
import Panel from 'react-toolbox/lib/layout/Panel';
import ReactTooltip from 'react-tooltip';
import { connect } from 'react-redux'
import { ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.min.css';
import Sidenav from '../components/sidenav/Sidenav';
import Header from '../components/header/Header'
import ShowOnMedia from '../components/ShowOnMedia'
import TaskQuickAccess from '../components/taskQuickAccess/TaskQuickAccess'
import { fetchProfile } from '../reducers/profile.reducer';
import './App.css';

class App extends Component {
  componentDidMount() {
    this.props.dispatch(fetchProfile());
  }

  render() {
    const { sidenav, children } = this.props;
    const containerStyle = {
      minHeight: '100vh'
    };
    if (sidenav.pinned) {
      containerStyle.paddingLeft = 256
    }
    return (
      <Layout>
        <Sidenav />
        <Panel style={containerStyle}>
          <ToastContainer autoClose={3000} position="bottom-right" />
          <Header />
          <ShowOnMedia mediaKey="small">
            <TaskQuickAccess />
          </ShowOnMedia>
          <main className="main">{children}</main>
        </Panel>
      </Layout>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    sidenav: state.sidenav
  }
}

export default connect(mapStateToProps)(App)
