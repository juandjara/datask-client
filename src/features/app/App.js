import React, { Component } from 'react'
import Layout from 'react-toolbox/lib/layout/Layout';
import Panel from 'react-toolbox/lib/layout/Panel';
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.min.css';

import {Sidenav} from '../../components/sidenav';
import {Header} from '../../components/header'
import ShowOnMedia from '../../components/ShowOnMedia'
import {TaskQuickAccess} from '../../components/taskQuickAccess'
import { reducer as profileReducer } from '../profile';
import './App.css';

class App extends Component {
  componentDidMount() {
    this.actions.fetchProfile()
  }

  render() {
    const { sidenav, children } = this.props;
    const containerStyle = {
      minHeight: '100vh'
    };
    if (sidenav.pinned) {
      containerStyle.paddingLeft = 256
    }
    return (
      <Layout>
        <Sidenav />
        <Panel style={containerStyle}>
          <ToastContainer autoClose={3000} position="bottom-right" />
          <Header />
          <ShowOnMedia mediaKey="small">
            <TaskQuickAccess />
          </ShowOnMedia>
          <main className="main">{children}</main>
        </Panel>
      </Layout>
    )
  }
}

const mapStateToProps = ({sidenav}) => ({sidenav})
const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({
    fetchProfile: profileReducer.fetchProfile
  }, dispatch)
})

export default connect(mapStateToProps, mapDispatchToProps)(App)
