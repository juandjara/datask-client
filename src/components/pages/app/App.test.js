import React from 'react'
import {App} from './App'
import renderer from 'react-test-renderer'
import {createMockStore} from 'redux-test-utils'
import {shallow} from 'enzyme'
import shallowWithStore from 'services/shallowWithStore'

let store;

describe('App', () => {
  beforeEach(() => {
    store = createMockStore({
      sidenav: {},
      profile: {},
      responsive: {}
    })
  })
  xit('matches snapshot', () => {
    const component = renderer.create(
      <App
        sidenav={{pinned: false}}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
  xit('matches snapshot with sidenav pinned', () => {
    const component = renderer.create(
      <App
        sidenav={{pinned: true}}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('calls fetchProfile on mount', () => {
    const component = shallowWithStore(
      <App
        sidenav={{pinned: false}}
        actions={{fetchProfile: jest.fn()}}
      />
    , store)
    const {props} = component.instance()
    expect(props.actions.fetchProfile).toHaveBeenCalled()
  })
  it('adds style when sidenav is pinned', () => {
    const component = shallow(
      <App
        sidenav={{pinned: false}}
        actions={{fetchProfile: jest.fn()}}
      />
    , {context: {store}})
    
    expect(component.childAt(1).props().style.paddingLeft).toBeUndefined()
    component.setProps({sidenav: {pinned: true}})
    component.update()
    expect(component.childAt(1).props().style.paddingLeft).toEqual(256)    
  })
})