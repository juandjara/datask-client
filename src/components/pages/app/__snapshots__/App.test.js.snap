// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<div
  className=""
>
  <div
    className="undefined"
    data-react-toolbox="panel"
    style={
      Object {
        "minHeight": "100vh",
      }
    }
  >
    <div />
    <header
      className="Header"
    >
      <h2
        className="Header-title"
      >
        Datask
      </h2>
    </header>
    <main
      className="main"
    />
  </div>
</div>
`;
